src/forms/RichTextEditor.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from "react";
+import ReactQuill from "react-quill";
+import "react-quill/dist/quill.snow.css";
+import "react-quill/dist/quill.bubble.css";
+
+const RichTextEditor = () => {
+    const modules = {
+        toolbar: [
+            [{ size: [false, "large"] }],
+            ["bold", "italic", "underline"],
+            ["blockquote", "code-block"],
+            [{ list: "ordered" }, { list: "bullet" }],
+            [
+                { indent: "-1" },
+                { indent: "+1" },
+                { align: ["", "center", "right", "justify"] },
+            ],
+            ["link", "image", "video"],
+            ["clean"],
+        ],
+    };
+
+    const formats = [
+        "header",
+        "size",
+        "bold",
+        "italic",
+        "underline",
+        "font",
+        "size",
+        "color",
+        "background",
+        "blockquote",
+        "code-block",
+        "list",
+        "bullet",
+        "indent",
+        "align",
+        "link",
+        "image",
+        "video",
+        "clean",
+    ];
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [value, setValue] = useState("");
+
+    return (
+        <ReactQuill
+            modules={modules}
+            formats={formats}
+            theme="snow"
+            onChange={setValue}
```

Comment:
We are using React-hook-form to handle the form state. This is the post content data, should we implement react-hook-form here or pass the data to onSubmit funtions in ComposePost and UpdatePost components?